fdbclient/include/fdbclient/BlobCipher.h

```diff
@@ -492,9 +519,11 @@ struct BlobCipherEncryptHeaderRef {
 	const EncryptHeaderCipherDetails getCipherDetails() const;
 	EncryptAuthTokenMode getAuthTokenMode() const;
 	EncryptCipherDomainId getDomainId() const;
+	EncryptHeaderCipherKCVs getKCVs() const;
 
 	void validateEncryptionHeaderDetails(const BlobCipherDetails& textCipherDetails,
```

Comment:
Yeah because by design BlobGranule does extra check for the file being decrypted by doing the header validation, add more datapoints, the header validation checks has evolved over period of time, so now decryption header checks does almost everything, however, BlobGranule check make sure GranuleCtxCipher details matches as expected. For now, I think we should leave it as is, it is light weight and more checks is useful, eventually I plan to move it under `isSimulated`

Reply:
Sure but maybe we should add a TODO then? If this is doing more checks than the decryptor we should definitely add these but maybe under `isSimulated`

Reply:
Actually looking more into the code, the only check repeated is `validation of KCV` , other checks are done to make sure the cipherKeys constructed by reading `BlobGranuleCipherKeysCtx` matches with the persisted header and that should remain as is. 
I think finding our if KCV is mismatch earlier is OK and not expensive enough to guard it with `isSimulation` tag, will leave it as is.

Reply:
Yeah let's add some sort of comment for this method at least so that if we add new encryption clients it's clear that calling this method is not necessary

Reply:
The method is named `validateEncryptionHeaderDetails`, hence, the goal is to validate the BlobCipherDetails and IVRef match between input details as well as persisted values. 

The checks during decryption are much more extensive including:
1. Validating details
2. Validate KCV
3. Validate Auth-tokens (if applicable)

Will add API documentation to list the details for better clarity